encoding/unstable_hex_stream.ts

```diff
@@ -2,96 +2,151 @@
 // This module is browser compatible.
 
 /**
- * Utilities for encoding and decoding to and from hex in a streaming manner.
+ * TransformStream classes to encode and decode to and from hexadecimal data in a streaming manner.
  *
  * ```ts
  * import { assertEquals } from "@std/assert";
- * import { HexDecoderStream } from "@std/encoding/unstable-hex-stream";
- * import { toText } from "@std/streams/to-text";
+ * import { encodeHex } from "@std/encoding/unstable-hex";
+ * import { HexEncoderStream } from "@std/encoding/unstable-hex-stream";
+ * import { toText } from "@std/streams";
  *
- * const stream = ReadableStream.from(["48656c6c6f2c", "20776f726c6421"])
- *   .pipeThrough(new HexDecoderStream())
- *   .pipeThrough(new TextDecoderStream());
+ * const readable = (await Deno.open("./deno.lock"))
+ *   .readable
+ *   .pipeThrough(new HexEncoderStream({ output: "string" }));
  *
- * assertEquals(await toText(stream), "Hello, world!");
+ * assertEquals(
+ *   await toText(readable),
+ *   encodeHex(await Deno.readFile("./deno.lock")),
+ * );
  * ```
  *
  * @experimental **UNSTABLE**: New API, yet to be vetted.
  *
  * @module
  */
 
-import { decodeHex, encodeHex } from "./hex.ts";
 import type { Uint8Array_ } from "./_types.ts";
 export type { Uint8Array_ };
+import {
+  calcMax,
```

Comment:
I missed this in #6480, but I don't think `calcMax` should be a part of public API. What do you think?

Reply:
It's a utility function for the raw encode functions. Without it people would need to be familiar with the ratio between the different bases.

The name though could be better. Essentially you'd use it to calculate the minimum size your buffer needs to be when passing it to encodeRawBase64